src/deephaven_ib/__init__.py

```diff
@@ -0,0 +1,894 @@
+
+        Returns:
+            RegisteredContract
+
+        Raises:
+              Exception
+        """
+
+        self._assert_connected()
+        cd = self._client.contract_registry.request_contract_details_blocking(contract)
+        return RegisteredContract(query_contract=contract, contract_details=cd)
+
+    def request_contracts_matching(self, pattern: str) -> Request:
+        """Request contracts matching a pattern.  Results are returned in the `contracts_matching` table.
+
+        Args:
+            pattern (str): pattern to search for.  Can include part of a ticker or part of the company name.
+
+        Returns:
+            Request
+
+        Raises:
+              Exception
+        """
+
+        self._assert_connected()
+        req_id = self._client.request_id_manager.next_id()
+        self._client.log_request(req_id, "MatchingSymbols", None, f"pattern='{pattern}'")
+        self._client.reqMatchingSymbols(reqId=req_id, pattern=pattern)
+        return Request(request_id=req_id)
+
+    ####################################################################################################################
+    ####################################################################################################################
+    ## Accounts
+    ####################################################################################################################
+    ####################################################################################################################
+
+    def request_account_pnl(self, account: str = "All", model_code: str = "") -> Request:
+        """Request PNL updates.  Results are returned in the `accounts_pnl` table.
+
+        Args:
+            account (str): Account to request PNL for.  "All" requests PNL for all accounts.
+            model_code (str): Model used to evaluate PNL.
+
+        Returns:
+            Request
+
+        Raises:
+              Exception
+        """
+
+        self._assert_connected()
+        req_id = self._client.request_account_pnl(account, model_code)
+        return Request(request_id=req_id)
+
+    ####################################################################################################################
+    ####################################################################################################################
+    ## News
+    ####################################################################################################################
+    ####################################################################################################################
+
+    def request_news_historical(self, contract: RegisteredContract, start: dtu.DateTime, end: dtu.DateTime,
+                                provider_codes: List[str] = None, total_results: int = 100) -> List[Request]:
+        """ Request historical news for a contract.  Results are returned in the `news_historical` table.
```

Comment:
I used the singular here because it is the action of requesting.  I was not intending it to refer to the return value.  This also ends up being consistent with the other method names.  e.g. all methods that request start with `request_`.